lib/vmpooler/pool_manager.rb

```diff
@@ -761,12 +799,7 @@ def execute!(maxloop = 0, loop_delay = 1)
         end
 
         $config[:pools].each do |pool|
-          if ! $threads[pool['name']]
-            check_pool(pool)
-          elsif ! $threads[pool['name']].alive?
-            $logger.log('d', "[!] [#{pool['name']}] worker thread died, restarting")
-            check_pool(pool)
-          end
+          evaluate_pool(pool, task_limit, maxloop, loop_delay)
```

Comment:
I think the first thing to do is reduce the vSphere connection count and then we can look at a Thread Pooler to properly handle queuing pool evaluations in a more sane manner. e.g. use min-max thread pool so we burst if multiple pools need something but use minimal resources when it's just doing nothing.